lib/apirequest.js

```diff
@@ -51,6 +51,32 @@ function getMissingParams(params, required) {
   return missing.length > 0 ? missing : null;
 }
 
+// Escaping path segement, please look at: http://blog.lunatech.com/2009/02/03/what-every-web-developer-must-know-about-url-encoding
+var pathSegmentEscapeChars = {
+  '#': '%23',
```

Comment:
@ssuda What did you reference when you implemented this? The blog post you linked to does not cover the more technical aspects of safely encoding the path params component.